File: packages/schemas/python/stress_test.py
"""
Network vulnerability stress-test schemas — Slice 6.

Mirrors the response shape returned by `POST /api/v1/network/stress-test`
(see `apps/api/src/network/stress_test.py`). Used by the
`/simulator/stress-test` page to render the heatmap + ranked-vulnerability
table.
"""
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel, Field


class ScenarioResult(BaseModel):
    """One Monte-Carlo iteration outcome."""
    airport:             str
    event_kind:          str
    severity:            str
    duration_min:        int
    affected:            int
    direct_hits:         int
    cascade_1:           int
    cascade_2:           int
    cancelled_estimate:  int
    total_delay_min:     int
    pax_delay_min:       int
    score:               float


class AirportSummary(BaseModel):
    """Roll-up over all iterations for a single airport."""
    airport:           str
    iterations:        int   = 0
    avg_affected:      float = 0.0
    p95_affected:      int   = 0
    avg_pax_delay_min: float = 0.0
    p95_pax_delay_min: int   = 0
    avg_score:         float = 0.0
    p95_score:         float = 0.0
    worst_kind:        str   = ""
    samples:           list[ScenarioResult] = Field(default_factory=list)


class StressTestRequest(BaseModel):
    airports:              Optional[list[str]] = None
    event_kinds:            Optional[list[str]] = None
    iterations_per_airport: int = 5
    seed:                   Optional[int] = 42


class StressTestResponse(BaseModel):
    iterations_per_airport: int
    total_scenarios:        int
    airports:               list[str]
    event_kinds:            list[str]
    ranked:                 list[AirportSummary] = Field(default_factory=list)
    # Heatmap: airport → event_kind → max score across iterations.
    heatmap:                dict[str, dict[str, float]] = Field(default_factory=dict)
    fleet_size:             int = 0
    schedule_size:          int = 0

File: packages/schemas/src/stress_test.ts
// ── Network vulnerability stress-test schemas (Slice 6) ──────────────────────
//
// Mirrors `apps/api/src/network/stress_test.py`. Consumed by the
// `/simulator/stress-test` page to render the heatmap + ranked-vulnerability
// table.

/** One Monte-Carlo iteration outcome. */
export interface ScenarioResult {
  airport:            string
  event_kind:         string
  severity:           string
  duration_min:       number
  affected:           number
  direct_hits:        number
  cascade_1:          number
  cascade_2:          number
  cancelled_estimate: number
  total_delay_min:    number
  pax_delay_min:      number
  score:              number
}

/** Roll-up over all iterations for a single airport. */
export interface AirportSummary {
  airport:           string
  iterations:        number
  avg_affected:      number
  p95_affected:      number
  avg_pax_delay_min: number
  p95_pax_delay_min: number
  avg_score:         number
  p95_score:         number
  worst_kind:        string
  samples:           ScenarioResult[]
}

export interface StressTestRequest {
  airports?:              string[]
  event_kinds?:            string[]
  iterations_per_airport?: number   // default 5, clamped [1, 20]
  seed?:                   number | null
}

export interface StressTestResponse {
  iterations_per_airport: number
  total_scenarios:        number
  airports:               string[]
  event_kinds:            string[]
  ranked:                 AirportSummary[]
  /** Heatmap matrix: airport → event_kind → max score across iterations. */
  heatmap:                Record<string, Record<string, number>>
  fleet_size:             number
  schedule_size:          number
}
